lib/src/install.rs

```diff
@@ -925,6 +925,38 @@ pub(crate) fn setup_tmp_mounts() -> Result<()> {
     Ok(())
 }
 
+#[context("Ensuring sys mounts")]
+pub(crate) fn setup_sys_mounts() -> Result<()> {
+    tracing::debug!("Setting up sys mounts");
+    // First of all, does efivars even exist in the host? If not, we are
+    // not dealing with an EFI system
+    let _ = rustix::fs::lstat("/proc/1/root/sys/firmware/efi/efivars")?;
+
+    // Now, let's find out if it's populated
+    let od = rustix::fs::open(
+        "/proc/1/root/sys/firmware/efi/efivars",
+        rustix::fs::OFlags::DIRECTORY,
+        rustix::fs::Mode::empty(),
+    )?;
+
+    let dir_populated = rustix::fs::Dir::read_from(od)?.next().is_some();
+    if dir_populated {
+        // This means the host has this mounted, so we should mount it too
+        tracing::debug!("mounting efivarfs");
+
+        // This looks to be a workaround for a problem elsewhere. Not sure if
+        // it's in podman, but just attempting to mount this is enough to trigger
+        // it being populated even when there is an existing mount.
+        // Given that, we will just try to mount and ignore the result.
```

Comment:
I don't quite understand the rationale behind ignoring the error.  Is it erroring out now?

Reply:
I will double-check tomorrow. But what I think I was seeing today was that when `bootc install to-filesystem --replace=alongside /target` was run... On the first invocation /sys/firmware/efi/efivars was unpopulated, but {/proc/1/root,/target}/sys/firmware/efi/efivars are populated, and the mount command fails, saying "efivarfs already mounted at /target/sys/firmware/efi/efivars". 

But once that was run, and failed out, "magically" /sys/firmware/efi/efivars was now populated, and the bootc run would complete without error.

I'm totally game with working on this further to make it more resilient. I also think this is some kind of bug elsewhere. I'm just not sure where to point the finger. 

So, my rationale here was that it didn't matter if the mount command itself succeeded, but just that it was run.

Reply:
FWIW I did verify that adding `-v /sys:/sys:rslave` to our `podman run` invocation gets us the efivars mount (as expected) alongside others that we also currently mount manually like `selinuxfs`.

>  and the mount command fails, saying "efivarfs already mounted at /target/sys/firmware/efi/efivars".

That's weird indeed. In a quick test I can't reproduce that with a simple

```
$ podman run --rm -ti --privileged --pid=host quay.io/fedora/fedora:39
# mount -t efivarfs efivars /sys/firmware/efi/efivars
```

Probably worth trying to dig into what's different about what we're doing here.

Reply:
The proper direction may be then to just do a doc change for this, while we investigate the origin of the error. 

Is it perhaps ARM specific?

Reply:
Also, I think the other key is that `-v /:/target` is used. As efivarfs will be mounted in /target.

And of course I'm now unable to reproduce this scenario. 🙄 

I'll update this patch with all of the suggestions and stop ignoring this error, and we can go from there.

Reply:
> Also, I think the other key is that -v /:/target is used. As efivarfs will be mounted in /target.

Ahh yes, tricky.  In the case that we have the full target rootfs, we could in our container simply do e.g. `mount --rbind /target/sys /sys` or so.

Actually for this "alongside" case it would definitely help there if we still didn't need to pass in the `-v /:/target` even and just got the root filesystem by jumping out of the mount namespace.